src/mem/cache/prefetch/bop.cc

```diff
@@ -201,27 +215,37 @@ BOP::bestOffsetLearning(Addr x)
         }
     }
 
+    // Move the offset iterator forward to prepare for the next time
     offsetsListIterator++;
 
     // All the offsets in the list were visited meaning that a learning
     // phase finished. Check if
     if (offsetsListIterator == offsetsList.end()) {
         offsetsListIterator = offsetsList.begin();
         round++;
+    }
+
+    // Check if its time to re-calculate the best offset
+    if ((bestScore >= scoreMax) || (round >= roundMax)) {
+        round = 0;
 
-        // Check if the best offset must be updated if:
-        // (1) One of the scores equals SCORE_MAX
-        // (2) The number of rounds equals ROUND_MAX
-        if ((bestScore >= scoreMax) || (round == roundMax)) {
+        // If the current best score (bestScore) has exceed the threshold to
+        // enable prefetching (badScore), reset the learning structures and
+        // enable prefetch generation
+        if (bestScore > badScore)  {
             bestOffset = phaseBestOffset;
             round = 0;
             bestScore = 0;
             phaseBestOffset = 0;
             resetScores();
             issuePrefetchRequests = true;
-        } else if (bestScore <= badScore) {
+        }
+        else {
```

Comment:
According to the guidelines the else must be in the same line as the }

Reply:
Fixed. By the way, is there a clang format file which outlines all the rules for Gem5 somewhere? I'm asking as right now I'm fixing the formatting manually because of which I sometimes miss some changes. With a clang-format file, all this work can be automated. Moreover, we can also add it to the commit hooks so that it automatically lints the code before committing